### ISSUE TITLE
Gap in Read and Write when HA Prometheus replica changes 

### ISSUE THREAD

**Issue by andreimiclea99** [grafana/mimir#7495]
#### Describe the bug
Every time the Prometheus replica changes in the HA tracker it leads to around 30 seconds gap in Mimir writes and Reads.
<img width="465" alt="image" src="https://github.com/grafana/mimir/assets/44814444/b08fd35c-28a2-4624-8a4c-1a120e4e4b13">

Another issue that I see when the Prometheus replica changes is that I see duplicated values for some metrics, for instance `count(count(container_memory_usage_bytes{namespace="$namespace",container="$container",pod=~"$pod"}) by (instance))`
with the output:
<img width="488" alt="image" src="https://github.com/grafana/mimir/assets/44814444/469cf247-16ff-4b91-94b6-afdb8d2f8f05">

The scenario when the HA Tracker replica changes is when the elected Prometheus pod gets terminated, either because node termination, OOM or simply because of pod deletion.

There is no data lost, but we have some sensitive alerts in production that triggers when something like this happens.
#### Environment
The Mimir and the two Prometheus are running inside the Kubernetes, the Mimir version is 2.11, notice the same behaviour on 2.9 and 2.10.
For deployment I used the mimir-distributed helm chart, version 5.0.0 .

#### Additional Context
Prometheus scrapping time is 30 seconds and when this happens i don't see any error logs or resources spikes in Mimir components.
Not sure if relevant but I am not using Memcached for caching.

**Comment by dimitarvdimitrov** [comment]
The dropped 30s of data and the duplicated series sound like [expected behaviour](https://grafana.com/docs/mimir/latest/configure/configure-high-availability-deduplication/). Have you tried tuning these three settings?
```
  # (advanced) Update the timestamp in the KV store for a given cluster/replica
  # only after this amount of time has passed since the current stored
  # timestamp.
  # CLI flag: -distributor.ha-tracker.update-timeout
  [ha_tracker_update_timeout: <duration> | default = 15s]

  # (advanced) Maximum jitter applied to the update timeout, in order to spread
  # the HA heartbeats over time.
  # CLI flag: -distributor.ha-tracker.update-timeout-jitter-max
  [ha_tracker_update_timeout_jitter_max: <duration> | default = 5s]

  # (advanced) If we don't receive any samples from the accepted replica for a
  # cluster in this amount of time we will failover to the next replica we
  # receive a sample from. This value must be greater than the update timeout
  # CLI flag: -distributor.ha-tracker.failover-timeout
  [ha_tracker_failover_timeout: <duration> | default = 30s]
```

**Comment by andreimiclea99** [comment]
@dimitarvdimitrov 

I forgot to mention that I changed those values with:
```
      ha_tracker:
        ha_tracker_update_timeout: 5s
        ha_tracker_update_timeout_jitter_max: 5s
        ha_tracker_failover_timeout: 12s
```

The above screenshots are with the above values, it indeed reduced the gap to around 30 seconds.
Before changing those values the gap was more than 60 seconds, so there was an improvement.

**Comment by dimitarvdimitrov** [comment]
I still think the lost scrape is the documented and expected behaviour. Do you see anything that's in the docs that doesn't match with what you observed?

There's also the grafana agent's clustering mode which doesn't rely on mimir's HA to failover, so it might be a bit less noticeable around failures https://grafana.com/docs/agent/latest/flow/concepts/clustering/

**Comment by andreimiclea99** [comment]
@dimitarvdimitrov To be honest I am not sure if this is the [expected behaviour](https://grafana.com/docs/mimir/latest/configure/configure-high-availability-deduplication/) for my case, but sounds like it.

Looks a bit better with these values:

```
distributor:
  extraArgs:
    distributor.ha-tracker.update-timeout-jitter-max: 2s
    distributor.ha-tracker.update-timeout: 2s
    distributor.ha-tracker.failover-timeout: 5s
```
Will try to do more tests.

What is a bit annoying is the fact that for e couple minutes it sees duplicated values for some metrics, when the Prometheus change happens.
